contracts/src/starknet/rewaave/token_bridge.cairo

```diff
@@ -70,17 +73,32 @@ func get_l1_token_bridge{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range
     return (res)
 end
 
+# Internals.
+
+func auth_governor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}():
+    let (caller_address) = get_caller_address()
+    let (governor_) = get_governor()
+    with_attr error_message("Called address should be {governor_}"):
+        assert caller_address = governor_
+    end
+    return ()
+end
+
+func auth_l1_handler{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(from_address_ : felt):
+    let (expected_from_address) = get_l1_token_bridge()
+    with_attr error_message("Expected deposit from l1_token_bridge: {expected_from_address}"):
+        assert from_address_ = expected_from_address
+    end
+    return ()
+end
+
 # Externals.
 
 @external
 func set_l1_token_bridge{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
         l1_bridge_address : felt):
     # The call is restricted to the governor.
-    let (caller_address) = get_caller_address()
-    let (governor_) = get_governor()
-    with_attr error_message("Called address should be {caller_address}"):
-        assert caller_address = governor_
-    end
+    auth_governor()
```

Comment:
Can we use OZs ownable?

Reply:
that's indeed the very same logic, we could only write `auth_governor()` using OZ, but not `auth_l1_handler()`. also we should rename `governor` to `owner` while the L1 counterpart is named `governor`. I would kep it that way for those reasons, but it's not a strong opinion. ok to use OZ if you think it's better

Reply:
that's indeed the very same logic, we could only write `auth_governor()` using OZ, but not `auth_l1_handler()`. also we should rename `governor` to `owner` while the L1 counterpart is named `governor`. I would kep it that way for those reasons, but it's not a strong opinion. ok to use OZ if you think it's better

Reply:
Happy to skip OZ then. Agree with the rename